### PR TITLE
[PBA-6208] Manchester United Limited (HQ UK) - iOS SDK] Video not set to full screen mode when a screen is rotated to landscape

### PR DESCRIPTION
**Pull request by IvanSakharovskii**

### CODE REVIEW

**Comment by brol1dev** [comment]
I don't think this one should be included in the current release. It will take time to QA to verify it and if other issues come up we will delay the release. I'll verify it and try to merge it for 4.34.0. release.

File: sdk/iOS/OoyalaSkinSDK/ViewControllers/FullscreenStateController/FullscreenStateController.m [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    
+    // Completion
+    
+    if (completion) {
+      completion();
+    }
+  }];
+}
+
+- (void)openInlineMode:(nullable void (^)())completion {
```
Comment (pilievOoyala) [inline sdk/iOS/OoyalaSkinSDK/ViewControllers/FullscreenStateController/FullscreenStateController.m:138]: We need docs for all these methods. I don't understand what this one does.

**Comment by pilievOoyala** [comment]
LGTM but we need documentation for the new methods.  We'll wait to merge this till later based on Eric's comment.

**Comment by DmitriiMaslov** [comment]
@rcvrgs Please check this PR . Tech support guys push us about this.

File: sdk/iOS/OoyalaSkinSDK/ViewControllers/FullscreenStateController/FullscreenStateController.m (hunk 1 above)
Comment (DmitriiMaslov) [reply sdk/iOS/OoyalaSkinSDK/ViewControllers/FullscreenStateController/FullscreenStateController.m:138]: I created s separate tickets that will be included into next sprint to write docs for new methods - https://jira.corp.ooyala.com/browse/PLAYER-3138

**Review by brol1dev — APPROVED** [review]
@pilievOoyala, @akshaydch11 the approach works fine. Checked the code and it looks ok.

Things to notice let product know that with this change the fullscreen animation is smoother but while the animation is happening playback is paused. This was previously supposed to happen but not handled accordingly, so it is fixed now.

Customers may come back saying that playback is getting paused while fullscreen is happening but we should just set the expectations correctly.

I'm approving, tested it locally. Feel free to merge it if you agree.

File: sdk/iOS/OoyalaSkinSDK/ViewControllers/FullscreenStateController/FullscreenStateOperation.m [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+@interface FullscreenStateOperation()
+
+@property (nonatomic, copy) void (^fullscreenBlock)(void (^animationCompletion)());
+@property (nonatomic, copy) void (^inlineBlock)(void (^animationCompletion)());
+@property (nonatomic, copy) void (^completeStateChanges)();
+@property (nonatomic) BOOL isFullscreen;
+
+// 'executing' and 'finished' exist in NSOperation, but are readonly
+@property (atomic, assign) BOOL _executing;
```
Comment (brol1dev) [inline sdk/iOS/OoyalaSkinSDK/ViewControllers/FullscreenStateController/FullscreenStateOperation.m]: remove the prefixed underscore.

File: sdk/iOS/OoyalaSkinSDK/ViewControllers/FullscreenStateController/FullscreenStateOperation.m [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+@property (atomic, assign) BOOL _finished;
```
Comment (brol1dev) [inline sdk/iOS/OoyalaSkinSDK/ViewControllers/FullscreenStateController/FullscreenStateOperation.m]: remove the prefixed underscore.

**Review by brol1dev — CHANGES_REQUESTED** [review]
added a comment, i'll change that on my side.

**Comment by brol1dev** [comment]
@IvanSakharovskii, @MarkEPAM, changed how you define the `executing` and `finished` properties in `FullscreenStateOperation.m` review it and test. Let us know if you agree or if you find problems with it. 

Tested it and it worked fine on my side.

@pilievOoyala, @akshaydch11, hold on merging this one. Let EPAM review the change I did too.

Thanks.

**Comment by IvanSakharovskii** [comment]
@rcvrgs @akshaydch11 Checked the changes. Looks fine. Think we can merge.

**Comment by akshaydch11** [comment]
Thanks @IvanSakharovskii  Merging it.

**Pull request merged**